pkg/client/image.go

```diff
@@ -144,17 +147,60 @@ func (c *client) ImagePush(ctx context.Context, imageName string, opts *ImagePus
 	return result, nil
 }
 
-func (c *client) ImageDelete(ctx context.Context, imageName string) (*apiv1.Image, error) {
+func (c *client) ImageDelete(ctx context.Context, imageName string, opts *ImageDeleteOptions) (*apiv1.Image, error) {
+	latest := ""
 	image, err := c.ImageGet(ctx, imageName)
 	if apierrors.IsNotFound(err) {
 		return nil, nil
 	} else if err != nil {
 		return nil, err
 	}
+	if err != nil {
+		return nil, err
+	}
+	if !((includesTag(imageName) || includesRepoAndTag(imageName)) && len(image.Tags) != 1) {
```

Comment:
Not blocking, but can we move the initialization of `latest` to here? Makes it clearer to understand its behavior at a glance.